Optimize the linear and log convex shape support function

jorisv opened the pull request:
This PR introduce the [highway framework](https://github.com/google/highway/tree/master)

# TODO

- [x] Add linear support function benchmark
- [x] Optimize highway implementation
- [x] Add log support function benchmark
- [x] Find log support function efficient implementation
- [x] Use SparseSet to improve visited node list clear
- [ ] Look at graph reordering algorithm
- [ ] Replace current implementation by optimized one

# Discussion

- [x] Should highway be an optional dependency: Yes
- [x] Should we hide highway from the public API: Yes
- [ ] Should we constraint the number of architectures ?

jorisv committed:
bench: Add convex_support.cpp benchmark

jorisv committed:
ci: Concatenate matrix.cxx_options to CXXFLAGS

jorisv committed:
bench: Add highway in convex_support benchmark

This commit had to wait highway integration in conda-forge: https://github.com/conda-forge/staged-recipes/pull/31092

jorisv committed:
pixi: Use conda-forge libhwy

ManifoldFR commented on benchmark/convex_support.cpp:
```diff
@@ -0,0 +1,356 @@
+}
+
+struct Icosahedron {
+  using TriangleIndex = std::tuple<std::size_t, std::size_t, std::size_t>;
```
Je pinaille, but `std::array<std::size_t, 3>` would be more appropriate here no?

jorisv committed:
bench: Run highway on double scalar type

jorisv committed:
bench: Use ILP to accelerate computation

jorisv committed:
bench: Fix ILP code and add 4 way test

jorisv committed:
bench: Refactor benchmark structure

- Move icosahedron implementation in a static library
- Add namespace
- Remove dead code

jorisv committed:
bench: Apply @ManifoldFR review

jorisv committed:
cmake: lint

jorisv committed:
ci: Fix Windows build

jorisv committed:
bench: Allocate per architecture to pad data structure

jorisv committed:
bench: Construct icosahedron neighbor and remove duplicated points

jorisv committed:
cmake: Simplify benchmark cmake

jorisv committed:
bench: Store points neighbors in a dedicated class

jorisv committed:
bench: Add log support function benchmark

jorisv committed:
bench: Remove SetSupportedTargetsForTest overhead

jorisv committed:
bench: Add WarmStartMesh data structure

jorisv committed:
Create algorithms from points instead of Icosahedron

jorisv committed:
bench: Don't store Icosahedron in IcosahedronWithNeighbors

jorisv committed:
bench: Add supportWithIndex highway implementation

jorisv committed:
bench: Add first SOA and Cluster algorithm

jorisv committed:
bench: Manage empty cluster and Add Log with Warm Start algorithm

jorisv committed:
pixi: Update lock file

jorisv committed:
bench: Improve LogWarmStart bench and remove some

jorisv committed:
bench: Add SOANeighborLogWarmStartAlgorithmBench

jorisv committed:
bench: Avoid doing too much allocation

jorisv committed:
bench: Add SOALocalNeighborLogWarmStartAlgorithm

jorisv committed:
bench: Add more standard deviation

jorisv committed:
bench: Use SparseSet instead of a std::vector
